src/prototype_room_external.js

```diff
@@ -390,26 +401,34 @@ Room.prototype.handleUnreservedRoom = function() {
       continue;
     }
 
-    let distance = Game.map.getRoomLinearDistance(this.name, roomName);
+    let distance = Game.map.getRoomLinearDistance(this.name, baseRoomName);
     if (distance > config.external.distance) {
       continue;
     }
 
-    let route = Game.map.findRoute(this.name, roomName);
+    let route = Game.map.findRoute(this.name, baseRoomName);
     distance = route.length;
     if (distance > config.external.distance) {
       continue;
     }
 
     if (room.memory.queue && room.memory.queue.length === 0) {
       let reservedRooms = _.filter(Memory.rooms, isReserved);
-      if (reservedRooms.length < room.controller.level - 1) {
+      /* RCL: target reserved rooms
+       * 4: 1
+       * 5: 3
+       * 6: 5
+       * 7: 7
+       * 8: 9
+       */
+      let numRoomsToReserve = (room.controller.level - 3) * 2 - 1;
+      if (reservedRooms.length < numRoomsToReserve) {
```

Comment:
The formula was recently introduced to sure if it is fine with the number of external rooms.

Currently I'm more thinking that the number of external rooms should depend on the number of spawns.
1 Spawn (RCL 1-6): 3 rooms
2 Spawns (RCL 7): 6 rooms
3 Spawns (RCL 8): 9 rooms

(Replace the number of rooms, with good numbers :-))